docs/internal.md

```diff
@@ -0,0 +1,72 @@
+# Internals of Azure SignalR Service
+
+- [Server Connections](#server-connections)
+- [Client Connections](#client-connections)
+- [Transport Data between Client and Server](#transport)
+
+Azure SignalR Service is built on top of ASP.NET Core SignalR framework.
+You can start from a local ASP.NET Core SignalR application, then seamlessly move to Azure SignalR Service with a few lines of code change.
+
+The following diagram shows the typical architecture when you use Azure SignalR Service with your own application server.
+Behavior of clients and servers will be different from self-hosting ASP.NET Core SignalR.
+More details are explained in below sections.
+
+![Architecture](./images/arch.png)
+
+## Server Connections
+
+When self-hosting ASP.NET Core SignalR, application server will listen for client connections and serve them directly.
+
+With SignalR Service, the application server is no longer responsible for accepting persistent client connections.
+A `negotiate` endpoint will be exposed by Azure SignalR Service SDK for each hub.
+It will respond to clients' negotiate requests and redirect them to SignalR Service.
+Eventually, all clients will be connected to SignalR Service.
+Read more about it in [Client Connections](#client-connections) section.
+
+After application server is started, the Azure SignalR SDK will open 5 WebSocket connections to the SignalR Service.
+The number of used connections is configurable as described at [here](./use-signalr-service.md#connectioncount).
+Messages from/to clients will be multiplexed into these connections.
+
+These connections are supposed to be connected to SignalR Service all the time.
+If a server connection is disconnected due to network issues,
+- all clients which are served by this server connection will be disconnected (read more about client and server pairing at [Transport Data between Client and Server](#transport) section);
+- the server connection will automatically start reconnecting.
+
+## Client Connections

```

Comment:
This is the implementation for ASP.NET Core SignalR, shall we point it out?

Reply:
Sure. Let me add it in the title.